### PR TITLE
feat(sol): integrated humanize-duration library to format duration

### PR DESCRIPTION
**Pull request by mostafakamar2308**
I also updated one line at `slot.test.ts` as it caused an error due to outdated types.

### CODE REVIEW

**Comment by neuodev** [comment]
🚀 Nova: https://nova-6nzd3rno9-litespace-cores-projects.vercel.app
🚀 Luna: https://luna-nnrgkyuz7-litespace-cores-projects.vercel.app
🚀 Nebula: https://nebula-ijrz7e9lb-litespace-cores-projects.vercel.app

**Comment by neuodev** [comment]
> I also updated one line at slot.test.ts as it caused an error due to outdated types.

Slots are deprecated. We replaced them with rules.

File: packages/sol/tests/duration.test.ts [hunk 1]
```diff
@@ -18,38 +18,42 @@ describe("Duration", () => {
...
-    expect(Duration.from("2min").formatShort()).to.be.eq("2m");
-    expect(Duration.from("0 mins").formatShort()).to.be.eq("");
-    expect(Duration.from("mins").formatShort()).to.be.eq("1m");
-    expect(Duration.from("د").formatShort()).to.be.eq("1m");
-    expect(Duration.from("1hr 30min").formatShort()).to.be.eq("1h 30m");
-    expect(Duration.from("1.5hr").formatShort()).to.be.eq("1h 30m");
-    expect(Duration.from("2.5hr").formatShort()).to.be.eq("2h 30m");
-    expect(Duration.from("1.hr").formatShort()).to.be.eq("1h");
-    expect(Duration.from(".5hr").formatShort()).to.be.eq("30m");
+  it("should format durations to EN", () => {
```
Comment (neuodev) [inline packages/sol/tests/duration.test.ts:44]:
```suggestion
  it("should format durations in English", () => {
```

File: packages/sol/tests/duration.test.ts [hunk 2]
```diff
@@ -18,38 +18,42 @@ describe("Duration", () => {
...
-    expect(Duration.from("2min").format()).to.be.eq("2 Minutes");
-    expect(Duration.from("0 mins").format()).to.be.eq("");
-    expect(Duration.from("mins").format()).to.be.eq("Minute");
-    expect(Duration.from("د").format()).to.be.eq("Minute");
-    expect(Duration.from("1hr 30min").format()).to.be.eq("Hour and 30 Minutes");
-    expect(Duration.from("1.5hr").format()).to.be.eq("Hour and 30 Minutes");
-    expect(Duration.from("2.5hr").format()).to.be.eq("2 Hours and 30 Minutes");
-    expect(Duration.from("1.hr").format()).to.be.eq("Hour");
-    expect(Duration.from(".5hr").format()).to.be.eq("30 Minutes");
+  it("should format durations to AR", () => {
```
Comment (neuodev) [inline packages/sol/tests/duration.test.ts:28]:
```suggestion
  it("should format durations in Arabic", () => {
```

File: packages/sol/tests/duration.test.ts [hunk 3]
```diff
@@ -18,38 +18,42 @@ describe("Duration", () => {
...
-    expect(Duration.from("0 mins").formatShort()).to.be.eq("");
-    expect(Duration.from("mins").formatShort()).to.be.eq("1m");
-    expect(Duration.from("د").formatShort()).to.be.eq("1m");
-    expect(Duration.from("1hr 30min").formatShort()).to.be.eq("1h 30m");
-    expect(Duration.from("1.5hr").formatShort()).to.be.eq("1h 30m");
-    expect(Duration.from("2.5hr").formatShort()).to.be.eq("2h 30m");
-    expect(Duration.from("1.hr").formatShort()).to.be.eq("1h");
-    expect(Duration.from(".5hr").formatShort()).to.be.eq("30m");
+  it("should format durations to EN", () => {
+    expect(Duration.from("").format("en")).to.be.eq("0 minutes");
```
Comment (neuodev) [inline packages/sol/tests/duration.test.ts:45]: Incase of `0` minutes. I think we should return empty string.

File: packages/luna/src/hooks/duration.ts
Comment (neuodev) [inline packages/luna/src/hooks/duration.ts:1]: Grate work. Please also remove these labels from `ar-EG.json`

**Review by neuodev — CHANGES_REQUESTED** [review]
Please make sure that all tests are passing.

File: packages/luna/src/hooks/duration.ts
Comment (mostafakamar2308) [reply packages/luna/src/hooks/duration.ts:1]: what labels?

**Comment by neuodev** [comment]
🚀 Nova: https://nova-kvva1nm08-litespace-cores-projects.vercel.app
🚀 Luna: https://luna-lrkx110bg-litespace-cores-projects.vercel.app
🚀 Nebula: https://nebula-daqdt1m4i-litespace-cores-projects.vercel.app

File: packages/luna/src/hooks/duration.ts
Comment (neuodev) [reply packages/luna/src/hooks/duration.ts:1]: https://github.com/litespace-org/litespace/pull/24#discussion_r1823169665

**Comment by neuodev** [comment]
🚀 Nova: https://nova-i0yn3jje0-litespace-cores-projects.vercel.app
🚀 Luna: https://luna-3v26y5enp-litespace-cores-projects.vercel.app
🚀 Nebula: https://nebula-36fa3kzso-litespace-cores-projects.vercel.app

File: pnpm-lock.yaml
Comment (neuodev) [inline pnpm-lock.yaml:1]: Why do we have a changes in the `pnpm-lock.yaml` file? Make sure not to introduce unrelated changes in your PRs.

**Comment by neuodev** [comment]
🚀 Nova: https://nova-ka4a5ozhj-litespace-cores-projects.vercel.app
🚀 Luna: https://luna-bfzwd14l1-litespace-cores-projects.vercel.app
🚀 Nebula: https://nebula-6bgxj1q9s-litespace-cores-projects.vercel.app

**Review by neuodev — APPROVED** [review]

**Pull request merged**